objectivec/google/protobuf/Type.pbobjc.m

```diff
@@ -29,6 +29,22 @@
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
 
+#pragma mark - Objective C Class references
+// This somewhat arcane code forces linkage of classes from static archives by
+// adding a concrete reference to the classes.
+// We don't use `[Foo class]` because we need a static value for our initializer.
+// This also has the added benefit of reducing size in that we don't have to
+// encode the class names and look them up at runtime.
+extern const struct {} GPBEnumValue_GPBOption_Class __asm__("_OBJC_CLASS_$_" GPBStringify(GPBOption));
+extern const struct {} GPBEnum_GPBEnumValue_Class __asm__("_OBJC_CLASS_$_" GPBStringify(GPBEnumValue));
+extern const struct {} GPBEnum_GPBOption_Class __asm__("_OBJC_CLASS_$_" GPBStringify(GPBOption));
+extern const struct {} GPBEnum_GPBSourceContext_Class __asm__("_OBJC_CLASS_$_" GPBStringify(GPBSourceContext));
+extern const struct {} GPBField_GPBOption_Class __asm__("_OBJC_CLASS_$_" GPBStringify(GPBOption));
```

Comment:
Why extern?  Can we make them static so they are scoped to this file?

So we actually fetch GPBOption three times here.  Rather than names them for the message class and then the field type, if we can scope them just to this file, why not use use the type, then you only need GPBOption once.

Reply:
Interestingly they need to be extern or the compiler crashes. I will be filing a bug with Apple. My next CL cleans up the repetition of the types.

Reply:
This is now fixed.